### ISSUE TITLE
Cannot save a blob

### ISSUE THREAD

**Issue by jkawamoto** [jimmywarting/StreamSaver.js#103]
After upgrading to v2, saving a blob doesn't work for me with Chrome 75.0.3770.80 on mac. [Saving a blob/file example](https://jimmywarting.github.io/StreamSaver.js/examples/saving-a-blob.html) also doesn't work, while the other examples work.

My code is something like this.
```js
import streamSaver from "streamsaver";

const stream = streamSaver.createWriteStream("output");
const writer = stream.getWriter();

await writer.ready;
await writer.write(new Blob(["abc"]));
await writer.close();
```

From my observation, the controller's [write](https://github.com/jimmywarting/StreamSaver.js/blob/5126612f58ca1ae3e050bbb06ff7bd412c390e86/StreamSaver.js#L260) and [close](https://github.com/jimmywarting/StreamSaver.js/blob/5126612f58ca1ae3e050bbb06ff7bd412c390e86/StreamSaver.js#L289) seem not to be called.

Is there any breaking changes between v1 and v2?

**Comment by jimmywarting** [comment]
Took some time to dubbug what was going wrong. 
saving a larger blob did work fine, so i was a bit in chock. 

Turns out StreamSaver closes the MessageChannel before it receives the downloadUrl from the service worker. It dose so when it's able to read the hole blob stream faster then it's able to load the mitm.html, install the service worker and receive the first initial postMessage coming from service worker.

https://github.com/jimmywarting/StreamSaver.js/blob/5126612f58ca1ae3e050bbb06ff7bd412c390e86/sw.js#L43-L45

https://github.com/jimmywarting/StreamSaver.js/blob/5126612f58ca1ae3e050bbb06ff7bd412c390e86/StreamSaver.js#L298-L299

---

Until there is a bug fix you could always do a setTimeout that wait's a little bit before it calls the destroyChannel (stream.close) 
```js
setTimeout(() => {
  blobStream.pipeTo(writableStream)
}, 1000)
```

At least until it receives the download url. or maybe call the createWriteStream earlier then you are able to get the blob that needs to be piped.

**Comment by jimmywarting** [comment]
The bigest differences from v1 and v2 is the core functions it uses.
It uses transferableStreams for starter and the codebase is almost completely refactored

the only public differences is that the second argument in `createWriteStream` takes an option object instead
```
    let opts = {
      size: null,
      pathname: null,
      writableStrategy: undefined,
      readableStrategy: undefined
    }
```

But i tried to make most of it as backwards  compatible as possible so no major breaking changes occur.

**Comment by jimmywarting** [comment]
also, you can't write a blob
```js
await writer.write(new Blob(["abc"]));
```

it has to be an Uint8array
```js
await writer.write(new Uint8Array([97, 98, 99])));
```

**Comment by jimmywarting** [comment]
fixed in 2.0.3

**Comment by jkawamoto** [comment]
I appreciate for fixing this quickly. My app and the example work perfectly with v2.0.3.

Btw, `await writer.write(new Blob(["abc"]));` was my bad. I use `Uint8Array` instead. 

Thanks!
